services/app/src/components/loginRoute/LoginRoute.jsx

```diff
@@ -1,8 +1,17 @@
 import React from 'react';
-import { Navigate, Outlet } from 'react-router';
+import { matchPath, Navigate, Outlet, useLocation } from 'react-router';
 import { useUserFromLocalStorage } from '../../hooks/useUserFromLocalStorage';
+import {useGameId} from "../../hooks/useGameId";
 
 const LoginRoute = () => {
+  const { pathname } = useLocation();
```

Comment:
I feel like this could be moved into the `Room` component, as you do not need to check if the route matches `game/:id` and also parse the pathname for the `gameId` as react router could handle that for us.

e.g.

```
const Room = () => {
  <...>
  const { id } = useParams();
  const { saveGameId } = useGameId();

  useEffect(() => {
     saveGameId(id);
  }, [id]);

  <...>
}


```

Reply:
this will not work, because user gets redirected to Login page befire Room page is rendered. Thats why store room id here and then redirect user back after login.